educe/stac/oneoff/cmd/weave.py

```diff
@@ -0,0 +1,200 @@
+# Author: Eric Kow
+# License: CeCILL-B (French BSD3-like)
+
+"""
+Inject annotations from an augmented corpus into the original.
+
+We only read the unannotated stage of the augmented corpus.
+It is augmented in the sense that it may contain extra bits
+of text interspersed with the original.
+"""
+
+from __future__ import print_function
+import copy
+import difflib
+import sys
+
+from educe.stac.oneoff.weave import\
+    (check_matches,
+     compute_updates,
+     shift_span)
+from educe.stac.util.args import\
+    (add_usual_input_args, add_usual_output_args,
+     get_output_dir, announce_output_dir,
+     read_corpus_with_unannotated)
+from educe.stac.util.output import save_document
+from educe.stac.util.doc import\
+    (evil_set_text,
+     compute_renames, rename_ids,
+     unannotated_key)
+from educe.util import mk_is_interesting
+import educe.stac
+
+
+def _preview_anno(doc, anno, max_width=50):
+    """Short text representation of an annotation"""
+    span = anno.text_span()
+    text = doc.text(span)
+    if len(text) > max_width:
+        snippet = text[:max_width] + '...'
+    else:
+        snippet = text
+    template = '{ty} {span} [{snippet}]'
+    return template.format(ty=anno.type,
+                           span=anno.text_span(),
+                           snippet=snippet)
+
+
+def _maybe_warn(warning, doc, annos):
+    """Emit a warning about a potentially problematic group of annotations
+    """
+    if annos:
+        oops = 'WARNING: ' + warning + ':\n'
+        oops += '\n'.join(['    {}'.format(_preview_anno(doc, x))
+                           for x in annos])
+        print(oops, file=sys.stderr)
+
+
+def _hollow_out_nonplayer_text(src_doc):
+    """Return a version of the source text where all characters in nonplayer
+    turns are replaced with a nonsense char (tab).
+
+    This is to prevent spurious matching by the sequence matcher.
+    """
+    # we can't use the API one until we update it to account for the
+    # fancy new identifiers
+    non_player_spans = [x.text_span() for x in src_doc.units
+                        if x.type == 'NonplayerTurn']
+    current = None
+    merged = []
```

Comment:
Maybe a less generic name like `merged_np_spans` would help to understand this code faster :-)